src/client/QXmppOmemoManager.cpp

```diff
@@ -0,0 +1,5446 @@
+/// QXmppTrustStorage *trustStorage = new QXmppTrustMemoryStorage;
+/// \endcode
+///
+/// A trust manager using its storage must be added to the client:
+/// \code
+/// QXmppTrustManager *trustManager = new QXmppAtmManager(trustStorage);
+/// client->addExtension(trustManager);
+/// \endcode
+///
+/// Afterwards, the OMEMO manager using its storage must be added to the client:
+/// \code
+/// QXmppOmemoManager *manager = new QXmppOmemoManager(omemoStorage);
+/// client->addExtension(manager);
+/// \endcode
+///
+/// You can set a security policy used by OMEMO.
+/// Is is recommended to apply TOAKAFA for good security and usability when
+/// using \xep{0450, Automatic Trust Management (ATM)}:
+/// \code
+/// manager->setSecurityPolicy(QXmpp::Toakafa);
+/// \endcode
+///
+/// \xep{0280, Message Carbons} should be used for delivering messages to all
+/// endpoints of a user:
+/// \code
+/// QXmppCarbonManager *carbonManager = new QXmppCarbonManager;
+/// client->addExtension(carbonManager);
+/// connect(client, &QXmppClient::connected, this, [=]() {
+///     carbonManager->setCarbonsEnabled(true);
+/// });
+/// connect(carbonManager, &QXmppCarbonManager::messageSent, manager,
+///     &QXmppOmemoManager::handleMessage);
+/// connect(carbonManager, &QXmppCarbonManager::messageReceived, manager,
+///     &QXmppOmemoManager::handleMessage);
+/// \endcode
+///
+/// The OMEMO data must be loaded before connecting to the server:
+/// \code
+///     manager->load();
+/// });
+/// \endcode
+///
+/// If no OMEMO data could be loaded (i.e., the result of \c load() is "false"),
+/// it must be set up first.
+/// That can be done as soon as the user is logged in to the server:
+/// \code
+/// connect(client, &QXmppClient::connected, this, [=]() {
+///     auto future = manager->start();
+/// });
+/// \endcode
+///
+/// Once the future is finished and the result is "true", the manager is ready
+/// for use.
+/// Otherwise, check the logging output for details.
+///
+/// By default, stanzas are only sent to devices having keys with the following
+/// trust levels:
+/// \code
+/// QXmpp::TrustLevel::AutomaticallyTrusted | QXmpp::TrustLevel::ManuallyTrusted
+/// | QXmpp::TrustLevel::Authenticated
+/// \endcode
+/// That behavior can be changed for each message being sent by specifying the
+/// accepted trust levels:
+/// \code
+/// QXmppSendStanzaParams params;
+/// params.setAcceptedTrustLevels(QXmpp::TrustLevel::Authenticated)
+/// client->send(stanza, params);
+/// \endcode
+///
+/// Stanzas can be encrypted for multiple JIDs which is needed in group chats:
+/// \code
+/// QXmppSendStanzaParams params;
+/// params.setEncryptionJids({ "alice@example.org", "bob@example.com" })
+/// client->send(stanza, params);
+/// \endcode
+///
+/// \warning THIS API IS NOT FINALIZED YET!
+///
+/// \ingroup Managers
+///
+/// \since QXmpp 1.5
+///
+
+struct DecryptionResult
+{
+    QDomElement serializedDecryptedSceContentElement;
```

Comment:
How about just `sceElement` (or maybe `sceContentElement`). At least "serialized" is already implied by `QDomElement`, right?